### PR TITLE
Add SIG Release Slack configs

### PR DESCRIPTION
**Pull request by justaugustus**
Supercedes: #3655 
/assign @calebamiles @tpepper 
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

### CODE REVIEW

**Comment by justaugustus** [comment]
cc: @hoegaarden @chrisz100 @vivektaparia @idealhack @SlickNik

File: communication/slack-config/restrictions.yaml [hunk 1]
```diff
@@ -13,4 +13,7 @@ restrictions:
   - path: "sig-docs/*.yaml"
     channels:
       - "^kubernetes-docs-[a-z]{2}$"
+  - path: "sig-release/*.yaml"
+    channels:
+      - ".*"
```
Comment (mrbobbytables) [inline communication/slack-config/restrictions.yaml]: I would be a bit more restrictive with the regexes here -- just to be a bit precautionary. 
```
- "^sig-release[a-z0-9-]{0,10}$"
- "^release-[a-z0-9-]{0,13}$"
```

**Comment by justaugustus** [comment]
@mrbobbytables -- thanks for the tip! Just cleaned up the regexes. :)

File: communication/slack-config/restrictions.yaml [hunk 2]
```diff
@@ -13,4 +13,8 @@ restrictions:
   - path: "sig-docs/*.yaml"
     channels:
       - "^kubernetes-docs-[a-z]{2}$"
+  - path: "sig-release/*.yaml"
+    channels:
+      - "^sig-release$"
+      - "^release-managers$"
```
Comment (mrbobbytables) [inline communication/slack-config/restrictions.yaml]: Sort of opposite from before XD Do you want them to be that restrictive? It means in the future you'd have to open it up or add another explicit entry.

Comment (justaugustus) [reply communication/slack-config/restrictions.yaml]: Yeah, that was the goal. For the time being, I don't want arbitrary groups slipping in. We can reassess later if it makes senses to fragment the usual conversations further.

Comment (mrbobbytables) [reply communication/slack-config/restrictions.yaml]: Cool. Just wanted to double check.

/lgtm

**Comment by Bubblemelon** [comment]
Thanks @justaugustus for making this PR! I've been swamped :smile:

**Pull request merged**